biocentral_server/predict/triton_predictor.py

```diff
@@ -0,0 +1,329 @@
+        }
+
+        # Connect to Triton
+        await self._connect_triton()
+
+        try:
+            # Convert embeddings to numpy and prepare for batching
+            embedding_ids = list(embeddings.keys())
+            embedding_arrays = [
+                embeddings[seq_id].cpu().numpy() for seq_id in embedding_ids
+            ]
+
+            # Process in batches
+            all_predictions = {}
+            for i in range(0, len(embedding_arrays), self.batch_size):
+                batch_embeddings = embedding_arrays[i : i + self.batch_size]
+                batch_ids = embedding_ids[i : i + self.batch_size]
+
+                # Pad embeddings to same length for batching
+                max_len = max(emb.shape[0] for emb in batch_embeddings)
+                embed_dim = batch_embeddings[0].shape[1]
+
+                padded_batch = np.zeros(
+                    (len(batch_embeddings), max_len, embed_dim), dtype=np.float32
+                )
+                for j, emb in enumerate(batch_embeddings):
+                    padded_batch[j, : emb.shape[0], :] = emb
+
+                # Call Triton
+                try:
+                    predictions = await self.triton_repo.compute_predictions(
+                        embeddings=padded_batch,
+                        model_name=self.triton_model_name,
+                    )
+
+                    # Post-process predictions
+                    batch_predictions = self._post_process_batch(
+                        predictions=predictions,
+                        embedding_ids=batch_ids,
+                    )
+                    all_predictions.update(batch_predictions)
+
+                except Exception as e:
+                    logger.error(
+                        f"Triton prediction failed for batch {i//self.batch_size}: {e}"
+                    )
+                    raise
+
+            return all_predictions
+
+        finally:
+            await self._disconnect_triton()
+
+    def predict(
+        self, sequences: Dict[str, str], embeddings: Dict[str, torch.Tensor]
+    ) -> Dict[str, List[Prediction]]:
+        """Run model prediction via Triton.
+
+        Args:
+            sequences: Dictionary mapping sequence hashes to amino acid sequences
+            embeddings: Dictionary mapping sequence hashes to embeddings
+
+        Returns:
+            Model predictions
+        """
+        # Run async prediction in event loop
+        loop = asyncio.new_event_loop()
+        asyncio.set_event_loop(loop)
+        try:
+            return loop.run_until_complete(
+                self._predict_async(sequences, embeddings)
+            )
+        finally:
+            loop.close()
+
+    def _post_process_batch(
+        self,
+        predictions: np.ndarray,
+        embedding_ids: List[str],
+    ) -> Dict[str, List[Prediction]]:
+        """Post-process Triton predictions into Prediction objects.
```

Comment:
Can this be unified with the base_model implementation?

Reply:
Done